### PR TITLE
added isMoneyTransferValid with Strings like exceptionhandler

### LINKED ISSUE #60: Implement error handling for Account API using global exception handling mechanism
Create the following custom exceptions:
InsufficientMoneyAmountException
UserNotFoundException
AccountNotFoundException
ExpiredAccountException

Rework Account API (controller and service) using a global exception handling mechanism.

### PR DESCRIPTION
**Pull request by andy-sam**
fixes #60

### CODE REVIEW

File: src/main/java/com/codesarge/banking/service/AccountService.java [hunk 1]
```diff
@@ -124,23 +128,25 @@ public Account findAccountById(Long id) {
...
-            return false;
-        }
+        if (accountFrom.getExpirationDate().isBefore(LocalDate.now()))
+            throw new ExpiredAccountException("account " + accountFrom.getId() + " is expired");
+
         //Checking if there enough funds in the account
-        return accountFrom.getBalance() >= requestMoneyTransferDto.getAmount();
+        if (accountFrom.getBalance() < requestMoneyTransferDto.getAmount())
+            throw new InsufficientMoneyAmountException(
+                    "insufficient funds to complete the transaction");
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/service/AccountService.java:149]: insufficient funds on + accountFrom.getId()

File: src/main/java/com/codesarge/banking/service/SubscriptionService.java [hunk 2]
```diff
@@ -41,10 +42,14 @@ public SubscriptionDto makeSubscription(SubscriptionRequestDto subscriptionReque
 
 //        checking if account money is enough for payment
         if (account.getBalance() < bankService.getPrice())
-            return null;
+            throw new InsufficientMoneyAmountException("Insufficient Money Amount") {
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/service/SubscriptionService.java:45]: "Insufficient Money Amount on account " + account.getId()

File: src/main/java/com/codesarge/banking/service/SubscriptionService.java [hunk 3]
```diff
@@ -54,7 +59,8 @@ public SubscriptionDto makeSubscription(SubscriptionRequestDto subscriptionReque
                 Periodicity periodicity = subscription.getBankService().getPeriodicity();
                 if ((periodicity.equals(Periodicity.WEEKLY) || periodicity.equals(
                         Periodicity.MONTHLY))) {
-                    return null;
+                    throw new IncorrectOperationException(
+                            "user is already subscribed to this recurring bank service");
```
Comment (alex-anakin) [inline src/main/java/com/codesarge/banking/service/SubscriptionService.java:63]: add subscription.getBankService().getName() to the log

**Review by alex-anakin — APPROVED** [review]

File: src/main/java/com/codesarge/banking/controllers/ExceptionAdvice.java [hunk 4]
```diff
@@ -30,4 +34,40 @@
...
         return new ResponseEntity<>(error, HttpStatus.NOT_FOUND);
     }
+
+    @ExceptionHandler(value = InsufficientMoneyAmountException.class)
+    public ResponseEntity<CustomErrorResponse> handleInsufficientMoneyAmountException(
+            InsufficientMoneyAmountException e) {
+        CustomErrorResponse error =
+                new CustomErrorResponse(e.getMessage());
+        error.setTimestamp(LocalDateTime.now());
+        return new ResponseEntity<>(error, HttpStatus.NOT_FOUND);
```
[the remark this answers is not in the document]
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/controllers/ExceptionAdvice.java:44]: done

File: src/main/java/com/codesarge/banking/service/AccountService.java (hunk 1 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/service/AccountService.java:149]: done

File: src/main/java/com/codesarge/banking/service/SubscriptionService.java (hunk 2 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/service/SubscriptionService.java:45]: done

File: src/main/java/com/codesarge/banking/service/SubscriptionService.java (hunk 3 above)
Comment (andy-sam) [reply src/main/java/com/codesarge/banking/service/SubscriptionService.java:63]: done

**Pull request merged**